code/modules/projectiles/projectile.dm

```diff
@@ -185,8 +224,9 @@
 				new_y--
 
 			step_towards(src, locate(new_x, new_y, z))
+			var/delay = 0

```

Comment:
Earlier it was declared and set in the same line.
Now I separated those lines because if it's hitscan it doesn't need a delay.